lib/webpacker/commands.rb

```diff
@@ -5,6 +5,27 @@ def initialize(webpacker)
     @webpacker = webpacker
   end
 
+  def clean
+    return if !config.public_output_path.exist? || !config.public_manifest_path.exist?
+
+    files_on_filesystem = Dir.glob("#{config.public_output_path}/**/*").select { |f| File.file? f }
+    files_in_manifest = manifest.refresh.values.map {|f| File.join config.root_path, 'public', f }
+
+    count_to_keep = 2
+    versions_to_keep = files_in_manifest.flat_map do |file_in_manifest|
+      file_prefix, file_ext = file_in_manifest.scan(/(.*)[0-9a-f]{20}(.*)/).first
+      versions_of_file = Dir.glob("#{file_prefix}*#{file_ext}").grep(/#{file_prefix}[0-9a-f]{20}#{file_ext}/)
```

Comment:
I've added the `grep` here to help filter out mismatches that `Dir.glob` might match. I'm thinking of something like `packs/users-abc123.js` and `packs/users-profile-def456.js`.

I'm trying to think of something that could still match. Maybe `users-baddadbabebaddadbabe-abc123.js` (i.e. the dev named the file something that matches the digest character space and length).